pkg/gateway/operations/operation_utils.go

```diff
@@ -32,6 +32,14 @@ func amzMetaWriteHeaders(w http.ResponseWriter, metadata catalog.Metadata) {
 	}
 }
 
+const amzMetadataDirectiveHeaderPrefix = "X-Amz-Metadata-Directive"
+
+// Per S3 API, if the header X-Amz-Metadata-Directive is set to 'REPLACE', the metadata should be replaced.
+// Otherwise, if the value is 'COPY' or the value is missing, it should be copied.
+func shouldCopyMetadata(req *http.Request) bool {
+	return !(req.Header.Get(amzMetadataDirectiveHeaderPrefix) == "REPLACE")
+}
+
```

Comment:
I think this makes it much more cleaner (and also expresses the default behavior):
```suggestion
func shouldReplaceMetadata(req *http.Request) bool {
	return req.Header.Get(amzMetadataDirectiveHeaderPrefix) == "REPLACE"
}

```